userapi/src/main/java/fk/prof/userapi/UserapiConfigManager.java

```diff
@@ -2,57 +2,39 @@
 
 import com.google.common.base.Preconditions;
 import com.google.common.io.Files;
+import io.vertx.core.DeploymentOptions;
 import io.vertx.core.json.JsonObject;
 
+import javax.validation.ConstraintViolation;
+import javax.validation.Validation;
+import javax.validation.Validator;
 import java.io.File;
 import java.io.IOException;
 import java.nio.charset.StandardCharsets;
 import java.util.Properties;
+import java.util.Set;
+import java.util.stream.Collectors;
 
 /**
- * Map configuration to POJO and encapsulate default settings inside it
- * Use this POJO to access config and remove jsonobject getters from rest of the code base
+ * Map configuration to {@link Configuration} and encapsulate default settings inside it.
  */
 public class UserapiConfigManager {
 
-  private static final String VERTX_OPTIONS_KEY = "vertxOptions";
-  private static final String USERAPI_HTTP_DEPLOYMENT_OPTIONS_KEY = "userapiHttpOptions";
+  private static Validator validator = Validation.buildDefaultValidatorFactory().getValidator();
+
   private static final String LOGFACTORY_SYSTEM_PROPERTY_KEY = "vertx.logger-delegate-factory-class-name";
   private static final String LOGFACTORY_SYSTEM_PROPERTY_DEFAULT_VALUE = "io.vertx.core.logging.SLF4JLogDelegateFactory";
-  private static final String STORAGE = "storage";
-  private static final String S3 = "s3";
-  private static final String THREAD_POOL = "thread.pool";
-  private static final String BUFFER_POOL_OPTIONS_KEY = "bufferPoolOptions";
-
-  static final String METRIC_REGISTRY = "backend-metric-registry";
-  private static final String PROFILE_RETENTION_KEY = "profile.retention.duration.min";
-  private static final String USERPAI_HTTP_PORT_KEY = "port";
-  private static final String REQ_TIMEOUT_KEY = "req.timeout";
-  private static final String CONFIG = "config";
 
-  private final JsonObject config;
+  public static final String METRIC_REGISTRY = "backend-metric-registry";
 
+  private final Configuration config;
 
   public UserapiConfigManager(String configFilePath) throws IOException {
     Preconditions.checkNotNull(configFilePath);
-    this.config = new JsonObject(Files.toString(
-        new File(configFilePath), StandardCharsets.UTF_8));
-  }
-
-  public UserapiConfigManager(JsonObject config) {
-    Preconditions.checkNotNull(config);
-    this.config = config;
-  }
-
-  JsonObject getVertxConfig() {
```

Comment:
Lets keep this around? Having this getter around makes sense since vertx options can be tweaked at any stage(testing/prod) without code changes.